cloud-platform/env-create.md

```diff
@@ -0,0 +1,232 @@
+
+```
+  #git clone the repo onto your local machine.
+  $ git clone git@github.com:ministryofjustice/k8s-nonprod-environments.git
+
+  # change directory into the k8s-non prod repo
+  cd k8s-nonprod-environments/
+
+  #create new branch in the repo, my-app is used as an example, you can call it something descriptive.
+  $ git checkout -b my-app
+
+  #you can confirm you are using the branch you made using the command.
+  $ git branch -a
+
+  #Current branch you are currently using will be indicated by the *.
+
+    master
+  * my-app
+    remotes/origin/HEAD -> origin/master
+    remotes/origin/add-git-crypt
+```
+
+2) Now we need to amend the namespaces directory in the root of repo and create an directory within it.
+
+```
+#we will need to change directories and change into the namespaces directory.
+cd namespaces/
+
+#we shall now be in our namespaces directory but we can confirm by the following command.
+pwd
+
+#output should show that we are in the namespaces directory. This will all be dependent on where you have chosen to git clone the repo. In this example k8s-nonprod was cloned into the Git directory which is located in Documents.
+
+$ pwd
+/Users/<username>/Desktop/Git/k8s-nonprod-environments/namespaces
+
+#Now we must create the directory for our service, the name of the directory we create should be in the format $servicename-$env
+$ mkdir myapp-dev
+
+#Now that we have created our namespaces subdirectory $servicename-env, lets now change into this directory.
+$ cd myapp-dev
+```
+
+3) Now we will need to create two files within our `$servicename-$env` directory.  `Namespaces.yaml` and `$servicename-$env-admin-role.yaml`. The structures of both files are shown below.
+
+
+#### k8s-non-prod-environment/namespaces/$servicename-$env/namespace.yaml
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: myapp-dev
+  labels:
+    name: myapp-dev
+```
+
+#### Namespace:
+
+Namespaces is a mechanism in Kubernetes that will essentially create your environment. 
+
+Kubernetes namespace creates a logical cluster within our cluster that provides isolation from any other namespace.
+
+Using a Kubernetes namespace could isolate namespaces for different environments in the same cluster. Providing us with flexibility of creating segregated environments for different.
+
+#### Namespace.yaml
+
+```
+apiVersion: 
+kind: 
+metadata:
+      name: This is where you will define your $servicename-$env
+      labels:
+           name: Also define your $servicename-$env
+```
+
+[https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/](https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/)
+
+#### k8s-non-prod-environment/namespaces/$servicename-$env/$service-$name-admin-role.yaml
+
+```
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: myapp-dev-admins
+  namespace: myapp-dev
+subjects:
+  - kind: Group
+    name: "github:$yourTeam"
```

Comment:
we need to explain what `$yourTeam` should be. It's their Github team name cast to a slug, so lowercased with spaces and underscores converted to dashes, or more simply the [last part of the URL when they view their team in Github](https://github.com/orgs/ministryofjustice/teams/digital-services).